src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ListViewModel.cs

```diff
@@ -7,21 +7,24 @@
 using CommunityToolkit.Mvvm.Input;
 using CommunityToolkit.Mvvm.Messaging;
 using Microsoft.CmdPal.Extensions;
+using Microsoft.CmdPal.Extensions.Helpers;
 using Microsoft.CmdPal.UI.ViewModels.Messages;
 using Microsoft.CmdPal.UI.ViewModels.Models;
 
 namespace Microsoft.CmdPal.UI.ViewModels;
 
 public partial class ListViewModel : PageViewModel
 {
-    private readonly HashSet<ListItemViewModel> _itemCache = [];
+    // private readonly HashSet<ListItemViewModel> _itemCache = [];
 
     // TODO: Do we want a base "ItemsPageViewModel" for anything that's going to have items?
 
     // Observable from MVVM Toolkit will auto create public properties that use INotifyPropertyChange change
     // https://learn.microsoft.com/dotnet/communitytoolkit/mvvm/observablegroupedcollections for grouping support
     [ObservableProperty]
-    public partial ObservableCollection<ListItemViewModel> Items { get; set; } = [];
+    public partial ObservableCollection<ListItemViewModel> FilteredItems { get; set; } = [];
+
+    public ObservableCollection<ListItemViewModel> Items { get; set; } = [];
```

Comment:
Isn't this just what the _itemCache was now? Fine with rename, just want to understand intent here. Should we make this private now?

Reply:
See I didn't trust: 
>  A set is a collection that contains no duplicate elements, and whose elements are in no particular order.

Order _does_ matter for these, at least it does in my head. If like, the Window Walker page wants to give us a list of windows in Z-order (which it determined), then we shouldn't muck with that at all. We should render those in _exactly that order_. I presumed that the `HashSet` would munge that ordering